### PR TITLE
DOC, MAINT: more distutils/build cleanups

### PR DESCRIPTION
**Pull request by tylerjereddy**
* remove some modules in `_build_utils` that don't seem relevant now that we use `meson` exclusively (it may be possible that a subset of the ideas get reused for ilp64 support in meson (?), but I suspect that will use meson idioms instead)

* update the detailed roadmap to remove the mention of removing support for `numpy.distutils` (it is true that there are still `meson` things to do of course, but `setup.py` is gone now)

* the paragraph about Pythran in the detailed roadmap was also deleted--it is no longer an optional dep

[skip cirrus]

### CODE REVIEW

File: doc/source/dev/roadmap-detailed.rst [hunk 1]
```diff
@@ -75,15 +67,6 @@ NumPy C API (by Cython, everything in SciPy was fixed) will disappear. Then we
 can define ``NPY_NO_DEPRECATED_API`` unconditionally.
```
Comment (ilayn) [inline doc/source/dev/roadmap-detailed.rst:67]: Actually this also happened. Not relevant but just saying.

Comment (rgommers) [reply doc/source/dev/roadmap-detailed.rst:67]: I think 99% there, but it's only fully true if we drop Cython 2.x and then update our use of `numpy_nodepr_api`.

File: scipy/_build_utils/_fortran.py [hunk 2]
```diff
@@ -1,445 +0,0 @@
...
-        'intelev': ['-i8'],  # Intel Visual Fortran Compiler for Itanium
-        'intelvem': ['-i8'],  # Intel Visual Fortran Compiler for 64-bit
-        'lahey': ['--long'],  # Lahey/Fujitsu Fortran 95 Compiler
-        'mips': ['-i8'],  # MIPSpro Fortran Compiler
-        'nag': ['-i8'],  # NAGWare Fortran 95 compiler
-        'nagfor': ['-i8'],  # NAG Fortran compiler
-        'pathf95': ['-i8'],  # PathScale Fortran compiler
-        'pg': ['-i8'],  # Portland Group Fortran Compiler
-        'flang': ['-i8'],  # Portland Group Fortran LLVM Compiler
-        'sun': ['-i8'],  # Sun or Forte Fortran 95 Compiler
```
Comment (rgommers) [inline scipy/_build_utils/_fortran.py:42]: This whole file seems fine to remove indeed - when we need it, we can dig it up from the 1.11.x branch.

**Review by rgommers — APPROVED** [review]
LGTM overall, thanks Tyler. Maybe restore the Pythran bit in the roadmap, updating it to mention `-Duse-pythran=false`, and then merge it?

**Comment by tylerjereddy** [comment]
Ok, I tried to restore/update the Pythran part of the roadmap.

**Review by rgommers — APPROVED** [review]
LGTM now, in it goes. Thanks Tyler, and thanks Ilhan for the review.

**Pull request merged**